boards/arm/nrf52840_mdk_m2_module/doc/index.rst

```diff
@@ -0,0 +1,38 @@
+.. _nrf52840_mdk_m2_module:
+
+nRF52840 MDK M.2 Module
```

Comment:
We do not add support for modules, it does not make sense. You can add support for [nrf52840-m2-developer-kit](https://makerdiary.com/products/nrf52840-m2-developer-kit)

Reply:
@jfischer-no The module is a standalone product that is being actively consumed by the community. The developer kit is a breakout board with no real functionality on its own, the value comes from the M2 card.

I think the Zephyr community would benefit from the developers being open to adding support for these boards, as it's a new form factor that's gaining some traction.

I'm willing to help on making the necessary changes to get this MR landed.

Reply:
> @jfischer-no The module is a standalone product that is being actively consumed by the community. The developer kit is a breakout board with no real functionality on its own, the value comes from the M2 card.
> 

No, [nrf52840-m2-developer-kit](https://makerdiary.com/products/nrf52840-m2-developer-kit) is a board with a DAPLink  adapter, Arduino headers and USB connector and can be easily integrated into Zephyr. A module is usually the SoC + some minimal circuit and the actual configuration changes depending on which board it is installed on. Of course, you can also approach it differently, insert m2_module as a board (btw, it does not meet the FCC requirements for modules, so I doubt anyone will benefit from it at all) and developer kit as a shield, but I assume that board_runner_args cannot be defined at shield level. Anyway, it is not convincing.

> I think the Zephyr community would benefit from the developers being open to adding support for these boards, as it's a new form factor that's gaining some traction.
> 
> I'm willing to help on making the necessary changes to get this MR landed.

"gaining some traction", come on, please do not exaggerate, https://docs.zephyrproject.org/latest/boards/arm/reel_board/doc/index.html#id4, and that was designed 3-4 years ago.